### ISSUE TITLE
Does akka support mqtt 5.0 client?

### ISSUE THREAD

**Issue by saumilsdk** [akka/alpakka#2622]
Does akka support mqtt 5.0 client?

**Comment by saumilsdk** [comment]
Hi @ennru when will akka mqtt paho client will support mqtt 5.0?

**Comment by ennru** [comment]
I'm not following the Paho MQTT community, but https://projects.eclipse.org/projects/iot.paho/releases/1.4.0-photon says there is MQTT 5.0 support in version 1.4.x.
Alpakka uses Paho version 1.2.5. I'm not aware of any work to upgrade that to later versions of Paho.

**Comment by saumilsdk** [comment]
> I'm not following the Paho MQTT community, but https://projects.eclipse.org/projects/iot.paho/releases/1.4.0-photon says there is MQTT 5.0 support in version 1.4.x.
> Alpakka uses Paho version 1.2.5. I'm not aware of any work to upgrade that to later versions of Paho.

Exactly! As you said Alpakka uses paho version 1.2.5 and again these alpakka extensions are community maintained, can you please gauge efforts required for the same and who is the expert to do this upgrade :)

**Comment by iosven** [comment]
I am very busy on the job, but if someone would upgrade https://doc.akka.io/docs/alpakka/current/mqtt.html accordingly I can at least offer to perform real world testing within days of being notified. Just ping me here on GitHub (and via email to sven DOT ludwig ATT nexocraft DOT com)

**Comment by iosven** [comment]
We are getting closer to the point where we would have to move away from alpakka unless this Issue is prioritized, because we will need MQTT 5.0 for sure.

**Comment by saumilsdk** [comment]
Truly bad support from lightband/akka team to take any ownership of alpakka mqtt module.

**Comment by ennru** [comment]
Please feel free to suggest the necessary changes to support MQTT 5.0.

**Comment by s1rc0** [comment]
any news ?

**Comment by ennru** [comment]
I had a look at the MQTT 5.0 story. The [Paho client 1.2.5](https://github.com/eclipse/paho.mqtt.java/releases/tag/v1.2.5) which Alpakka is using has two separate libraries and APIs for MQTT 4 and MQTT 5.
This is not a matter of a simple upgrade, but more about offering an additional API to talk MQTT 5.

I encourage the community again to dig into it and suggest a way to bring in MQTT 5 support.

**Comment by s1rc0** [comment]
@ennru  thanks!
a lot of people waiting for mqtt_v5 Akka Source, especially IoT guys

**Comment by s1rc0** [comment]
AKKA has already become paid, a million dollars have flowed in and there is still no mqtt_v5 support ? :)
